### PR TITLE
CORE-17383 Add new token selection worker

### PR DESCRIPTION
**Pull request by owenstanford**
Adds a new worker to host the Token Selection Processor.

Shared pipeline PR: https://github.com/corda/corda-shared-build-pipeline-steps/pull/1228

### CODE REVIEW

File: .ci/JenkinsfileCombinedWorkerPluginsSmokeTests [hunk 1]
```diff
@@ -1,4 +1,4 @@
-@Library('corda-shared-build-pipeline-steps@5.1') _
+@Library('corda-shared-build-pipeline-steps@owen/CORE-17383-Add-token-selection-worker') _
```
Comment (vkolomeyko) [inline .ci/JenkinsfileCombinedWorkerPluginsSmokeTests:1]: We probably need to undo those changes once `corda-shared-build-pipeline` is merged.

File: applications/workers/release/token-selection-worker/build.gradle [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        exclude group: 'org.osgi'
+    }
+
+    // This puts the jdbc driver into the docker image in the /opt/jdbc-driver folder
+    // this folder can contain many jdbc drivers (and DataSourceFactory provider bundles).
+    // Postgres doesn't need a DataSourceFactory provider bundle (e.g. pax-jdbc), because
+    // the postgres devs have written their own and it's in this jar (PGDataSourceFactory).
+    dockerImageJdbc "org.postgresql:postgresql:$postgresDriverVersion"
+
+    // If we were to do this for a different database that is *not natively an OSGi bundle*
```
Comment (vkolomeyko) [inline applications/workers/release/token-selection-worker/build.gradle:58]: Do we need those comments?

**Review by vkolomeyko — COMMENTED** [review]
Looks broadly fine.

We probably need `corda-api` PR as well to introduce this Worker. At least to specify which topics this worker will be consuming from and producing to.

File: .ci/JenkinsfileCombinedWorkerPluginsSmokeTests (hunk 1 above)
Comment (owenstanford) [reply .ci/JenkinsfileCombinedWorkerPluginsSmokeTests:1]: yes we will, we have to merge with these first, then merge the pipeline changes, then revert these refs back in a follow-up PR

File: applications/workers/release/token-selection-worker/build.gradle (hunk 2 above)
Comment (owenstanford) [reply applications/workers/release/token-selection-worker/build.gradle:58]: They are added every worker that uses the DB so I think it makes sense to keep them

**Review by jacob-scott3141 — CHANGES_REQUESTED** [review]
LGTM - will approve once Jenkinsfiles are reverted

**Comment by owenstanford** [comment]
> LGTM - will approve once Jenkinsfiles are reverted

@jacob-scott3141 
We need to merge this as-is, then merge the pipeline changes, then revert the jenkins changes back to 5.1 otherwise we will break the CI builds.

**Comment by owenstanford** [comment]
> Looks broadly fine.
> 
> We probably need `corda-api` PR as well to introduce this Worker. At least to specify which topics this worker will be consuming from and producing to.

We can do that after this goes in

**Review by vkolomeyko — APPROVED** [review]
LGTM

**Review by jacob-scott3141 — APPROVED** [review]

**Pull request merged**